src/SouffleInterface.h

```diff
@@ -303,6 +303,8 @@ class SouffleProgram {
         return allRelations;
     }
 
+    virtual void executeSubroutine(std::string name, const std::vector<RamDomain>* args,
```

Comment:
Would it make sense to use references instead of pointers for any of these? That lessens the need/temptation of raw pointers, for similar performance.